server/manager/src/main/java/org/apache/accumulo/manager/tableOps/compact/CompactionDriver.java

```diff
@@ -82,6 +82,14 @@ public long isReady(long tid, Manager manager) throws Exception {
           TableOperation.COMPACT, TableOperationExceptionType.OTHER, "Compaction canceled");
     }
 
+    String deleteMarkerPath = Constants.ZROOT + "/" + manager.getInstanceID() + Constants.ZTABLES
+        + "/" + tableId + Constants.ZTABLE_DELETE_MARKER;
+    if (zoo.exists(deleteMarkerPath, null)) {
```

Comment:
What's the behavior with ZK exists if the parent path of deleteMarkerPath is gone? Does it just return false, or does it throw an error? If it returns false, we may want to handle that.

Reply:
It would appear that `ZooKeeper.exists` returns `null` when a NONODE exception is returned by the ZK Server to the client. ZooReader returns false when null is returned.